Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeInformation.ps1

```diff
@@ -26,167 +26,84 @@ function Get-ExchangeInformation {
-                $netFrameworkExchange.MinSupportedVersion = [HealthChecker.NetMajorVersion]::Net4d6d2
-                $netFrameworkExchange.MaxSupportedVersion = [HealthChecker.NetMajorVersion]::Net4d6d2
-            } elseif ($buildInformation.CU -lt [HealthChecker.ExchangeCULevel]::CU21) {
-                $netFrameworkExchange.MinSupportedVersion = [HealthChecker.NetMajorVersion]::Net4d6d2
-                $netFrameworkExchange.MaxSupportedVersion = [HealthChecker.NetMajorVersion]::Net4d7d1
-            } elseif ($buildInformation.CU -lt [HealthChecker.ExchangeCULevel]::CU23) {
-                $netFrameworkExchange.MinSupportedVersion = [HealthChecker.NetMajorVersion]::Net4d7d1
-                $netFrameworkExchange.MaxSupportedVersion = [HealthChecker.NetMajorVersion]::Net4d7d2
-            } else {
-                $netFrameworkExchange.MinSupportedVersion = [HealthChecker.NetMajorVersion]::Net4d7d2
-                $netFrameworkExchange.MaxSupportedVersion = [HealthChecker.NetMajorVersion]::Net4d8
-            }
+        try {
+            $getOwaVirtualDirectory = Get-OwaVirtualDirectory -Identity ("{0}\owa (Default Web Site)" -f $Server) -ADPropertiesOnly -ErrorAction Stop
+            $getWebServicesVirtualDirectory = Get-WebServicesVirtualDirectory -Server $Server -ErrorAction Stop
+        } catch {
+            Write-Verbose "Failed to get OWA or EWS virtual directory"
+            Invoke-CatchActions
         }
 
-        $exchangeInformation.ExchangeEmergencyMitigationService = Get-ExchangeEmergencyMitigationServiceState `
-            -RequiredInformation ([PSCustomObject]@{
+        $params = @{
+            RequiredInformation = [PSCustomObject]@{
                 ComputerName       = $Server
                 MitigationsEnabled = if ($null -ne $PassedOrganizationInformation.OrganizationConfig) { $PassedOrganizationInformation.OrganizationConfig.MitigationsEnabled } else { $null }
-                GetExchangeServer  = $exchangeInformation.GetExchangeServer
-            }) `
-            -CatchActionFunction ${Function:Invoke-CatchActions}
+                GetExchangeServer  = $getExchangeServer
+            }
+            CatchActionFunction = ${Function:Invoke-CatchActions}
+        }
+
+        $exchangeEmergencyMitigationService = Get-ExchangeEmergencyMitigationServiceState @params
 
-        if (($OSMajorVersion -ge [HealthChecker.OSServerVersion]::Windows2016) -and
-            ($buildInformation.ServerRole -ne [HealthChecker.ExchangeServerRole]::Edge)) {
-            $exchangeInformation.AMSIConfiguration = Get-ExchangeAMSIConfigurationState -GetSettingOverride $PassedOrganizationInformation.SettingOverride
+        if (($windows2016OrGreater) -and
+        ($getExchangeServer.IsEdgeServer -eq $false)) {
+            $amsiConfiguration = Get-ExchangeAMSIConfigurationState -GetSettingOverride $PassedOrganizationInformation.SettingOverride
```

Comment:
```suggestion
            [array]$amsiConfiguration = Get-ExchangeAMSIConfigurationState -GetSettingOverride $PassedOrganizationInformation.SettingOverride
```

Same here. This was an `array` before. Otherwise the `.count()` method is not available.

![image](https://user-images.githubusercontent.com/40993616/217830517-8644f569-8956-4b96-ab6e-2535f6f164f3.png)